R/scope.R

```diff
@@ -347,17 +347,15 @@ renv_scope_trace <- function(what, tracer, envir = parent.frame()) {
 
 }
 
-renv_scope_var <- function(key, value, frame, envir = parent.frame()) {
 
-  if (exists(key, envir = frame, inherits = FALSE)) {
-    saved <- get(key, envir = frame, inherits = FALSE)
-    assign(key, value, envir = frame, inherits = FALSE)
-    defer(assign(key, saved, envir = frame, inherits = FALSE), envir = envir)
+renv_scope_binding <- function(envir, symbol, replacement, frame = parent.frame()) {
```

Comment:
I feel like we need different names here -- usually `envir` is used in these scope functions to mean something like "stack frame".

Reply:
That's why I originally had:

```suggestion
renv_scope_binding <- function(frame, symbol, replacement, envir = parent.frame()) {
```

but maybe some name other than `frame` is better here -- `container`? Something else?

Reply:
Or perhaps all of these `scope` functions should accept an argument called `scope` instead of `envir` just to remove some of the naming ambiguity?

Reply:
Yeah, I'll do another PR that renames to scope. 

The problem with the original names is that `envir` is the call frame and `frame` is the environment to modify, so they're perfectly wrong 😄

Reply:
But I didn't mean to do that change in this PR, so I'll remove it.